### PR TITLE
#18229 Setting TrustSelfSignedStrategy to avoid validating ssl certificates by default

### PR DESCRIPTION
**Pull request by nollymar**
With this change, we don't validate tls certificates. It is equivalent to using the `--insecure` flag when running from curl

### CODE REVIEW

File: dotCMS/src/main/java/com/dotcms/content/elasticsearch/util/DotRestHighLevelClientProvider.java [hunk 1]
```diff
@@ -106,14 +110,24 @@ private BasicCredentialsProvider getCredentialsProvider() throws IOException, Ge
...
+        final String esProtocol = Config.getStringProperty("ES_PROTOCOL", "https");
         final RestClientBuilder clientBuilder = RestClient
                 .builder(new HttpHost(Config.getStringProperty("ES_HOSTNAME", "127.0.0.1"),
-                        Config.getIntProperty("ES_PORT", 9200), sslContextFromPem != null ? "https" :"http"))
+                        Config.getIntProperty("ES_PORT", 9200), esProtocol))
                 .setHttpClientConfigCallback((httpClientBuilder) -> {
                     if (sslContextFromPem != null) {
                         httpClientBuilder
                                 .setSSLContext(sslContextFromPem);
+                    } else if ("https".equals(esProtocol)){
```
Comment (freddyucv) [inline dotCMS/src/main/java/com/dotcms/content/elasticsearch/util/DotRestHighLevelClientProvider.java:121]: "https" should be a constant

File: dotCMS/src/main/java/com/dotcms/content/elasticsearch/util/DotRestHighLevelClientProvider.java [hunk 2]
```diff
@@ -106,14 +110,24 @@ private BasicCredentialsProvider getCredentialsProvider() throws IOException, Ge
...
+                        try {
+                            httpClientBuilder.setSSLContext(SSLContexts
+                                    .custom().loadTrustMaterial(new TrustSelfSignedStrategy()).build());
+                        } catch (NoSuchAlgorithmException | KeyManagementException | KeyStoreException e) {
+                            Logger.error(this.getClass(),
+                                    "Error setting TrustSelfSignedStrategy for Elastic RestHighLevelClient",
+                                    e);
+                            throw new DotRuntimeException(e);
+                        }
```
Comment (freddyucv) [inline dotCMS/src/main/java/com/dotcms/content/elasticsearch/util/DotRestHighLevelClientProvider.java:130]: is it possible do any test?

Comment (nollymar) [reply dotCMS/src/main/java/com/dotcms/content/elasticsearch/util/DotRestHighLevelClientProvider.java:130]: how do you suggest to test it? The logic avoid sending certificates for validation when the ES connection is established

**Review by freddyucv — APPROVED** [review]

File: dotCMS/src/main/java/com/dotcms/content/elasticsearch/util/DotRestHighLevelClientProvider.java (hunk 1 above)
Comment (nollymar) [reply dotCMS/src/main/java/com/dotcms/content/elasticsearch/util/DotRestHighLevelClientProvider.java]: Done

**Review by dsilvam — APPROVED** [review]

**Review by jdotcms — APPROVED** [review]

**Pull request merged**